src/main/java/baritone/process/FarmProcess.java

```diff
@@ -96,7 +101,24 @@ public boolean isActive() {
     }
 
     @Override
-    public void farm() {
+    public void farm(int range, int posX, int posY, int posZ) {
+        //Reassigning values to incoming parameters is not recommended, therefor I Use temporary local variables instead.
+        int _posX = posX;
+        int _posY = posY;
+        int _posZ = posZ;
+        //0 for getting player pos
+        if (_posX == 0)
+            _posX = (int)ctx.playerFeetAsVec().x;
+        if (_posY == 0)
```

Comment:
using 0 like this is fucking awful, you could be standing at 0 or have a farm centered at 0?? (plausible in skyblock etc)

use optionalint for x y and z

however i could be convinced for -1 meaning unlimited range, since this is already a common pattern